specs/language/introduction.tex

```diff
@@ -0,0 +1,165 @@
+adjacent pixels and \gls{quad} operations allow passing data between adjacent
+lanes. In compute shaders quads may be one or two dimensional depending on the
+workload dimensionality described in the \texttt{numthreads} attribute on the
+entry function. (FIXME: Add reference to attribute)
+
+\Sub{\gls{threadgroup}}{Intro.Model.Group}
+
+\p A grouping of \gls{wave}s executing the same shader to produce a combined
+result is called a \gls{threadgroup}. \gls{threadgroup}s are executed on
+separate \acrshort{simd} hardware and are not instruction locked with other
+\gls{threadgroup}s.
+
+\Sub{\gls{dispatch}}{Intro.Model.Dispatch}
+
+\p A grouping of \gls{threadgroup}s which represents the full execution of a
+\acrshort{hlsl} program and results in a completed result for all input data
+elements.
+
+\Sec{\acrshort{hlsl} Memory Models}{Intro.Memory}
+
+\p Memory accesses for \gls{sm} 5.0 and earlier operate on 128-bit slots aligned
+on 128-bit boundaries. This optimized for the common case in early shaders where
+data being processed on the GPU was usually 4-element vectors of 32-bit data
+types.
+
+\p On modern hardware memory access restrictions are loosened, and reads of
+32-bit multiples are supported starting with \gls{sm} 5.1 and reads of 16-bit
+multiples are supported with \gls{sm} 6.0. \gls{sm} features are fully
+documented in the \gls{dx} Specifications, and this document will not attempt to
+elaborate further.
+
+\Sec{Common Definitions}{Intro.Defs}
+
+\p The following definitions are consistent between \acrshort{hlsl} and the
+\gls{isoC} and \gls{isoCPP} specifications, however they are included here for
+reader convenience.
+
+\Sub{Diagnostic Message}{Intro.Defs.Diags}
+\p An implementation defined message belonging to a subset of the
+implementation's output messages which communicates diagnostic information to
+the user.
+
+\Sub{Ill-formed Program}{Intro.Defs.IllFormed}
+\p A program that is not well formed, for which the implementation is expected
+to return unsuccessfully and produce one or more diagnostic messages.
+
+\Sub{Implementation-defined Behavior}{Intro.Defs.ImpDef}
+\p Behavior of a well formed program and correct data which may vary by the
+implementation, and the implementation is expected to document the behavior.
+
+\Sub{Implementation Limits}{Intro.Defs.ImpLimits}
+\p Restrictions imposed upon programs by the implementation.
+
+\Sub{Undefined Behavior}{Intro.Defs.Undefined}
+
+\p Behavior of invalid program constructs or incorrect data which this standard
+imposes no requirements, or does not sufficiently detail.
+
+\Sub{Unspecified Behavior}{Intro.Defs.Unspecified}
+\p Behavior of a well formed program and correct data which may vary by the
+implementation, and the implementation is not expected to document the behavior.
+
+\Sub{Well-formed Program}{Intro.Defs.WellFormed}
+\p An HLSL program constructed according to the syntax rules, diagnosable
+semantic rules, and the One Definition Rule.
```

Comment:
Just a note for future reference that we may want to come back to these and purge any we don't reference if there's any chance that might be the case.